probe-rs/src/debug/debug_info.rs

```diff
@@ -540,23 +539,37 @@ impl DebugInfo {
         let mut unwind_registers = initial_registers;
 
         // Unwind [StackFrame]'s for as long as we can unwind a valid PC value.
-        'unwind: while let Some(frame_pc_register_value) = unwind_registers
-            .get_program_counter()
-            .and_then(|pc| pc.value)
+        'unwind: while let Some(frame_pc_register_value) =
+            unwind_registers.get_program_counter().and_then(|pc| {
+                if pc.is_zero() | pc.is_max_value() {
+                    None
+                } else {
+                    pc.value
+                }
```

Comment:
```suggestion
                (pc.is_zero() | pc.is_max_value()).and_then(|| pc.value)
```

Reply:
Should be possible. maybe it needs an argument in the closure, not sure.

Reply:
The problem is that you can't put an `and_then` on a `bool`.